src/ResourceManager/Network/Commands.Network/NetworkWatcher/ConnectionMonitor/SetAzureNetworkWatcherConnectionMonitorCommand.cs

```diff
@@ -0,0 +1,262 @@
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipeline = true,
+            HelpMessage = "Connection monitor object.",
+            ParameterSetName = "SetByInputObject")]
+        [ValidateNotNull]
+        public PSConnectionMonitorResult InputObject { get; set; }
+
+        [Alias("ConnectionMonitorName")]
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The connection monitor name.",
+            ParameterSetName = "SetByName")]
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The connection monitor name.",
+            ParameterSetName = "SetByResource")]
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The connection monitor name.",
+            ParameterSetName = "SetByLocation")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The ID of the connection monitor source.")]
+        [ValidateNotNullOrEmpty]
+        public string SourceResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Monitoring interval in seconds.")]
+        [ValidateNotNullOrEmpty]
+        public int? MonitoringIntervalInSeconds { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Source port.")]
+        [ValidateNotNull]
+        [ValidateRange(1, int.MaxValue)]
+        public int SourcePort { get; set; }
+
+        [Parameter(
+             Mandatory = false,
+             HelpMessage = "The ID of the connection monitor destination.")]
+        [ValidateNotNullOrEmpty]
+        public string DestinationResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The Ip address of the connection monitor destination.")]
+        [ValidateNotNullOrEmpty]
+        public string DestinationAddress { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Destination port.")]
+        [ValidateNotNull]
+        [ValidateRange(1, int.MaxValue)]
+        public int DestinationPort { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Configure connection monitor, but do not start it")]
+        [ValidateNotNullOrEmpty]
+        public SwitchParameter ConfigureOnly { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "A hashtable which represents resource tags.")]
+        public Hashtable Tag { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        public string resourceGroupName = string.Empty;
+        public string networkWatcherName = string.Empty;
+        public string connectionMonitorName;
```

Comment:
@irrogozh same comment about moving these values below `base.Execute()` and setting them to the parameter values

Reply:
Done